src/components/Modal/IdentityConnectModal/AccountsConnectModal.tsx

```diff
@@ -0,0 +1,370 @@
+  currentStep: number
+  isTimerActive: boolean
+  isValidating: boolean
+  stepsCurrentHelper?: Record<number, string>
+}
+
+type CardHelperKeys = {
+  start: string
+  active: string
+  success: string
+  error?: string
+}
+
+const STEPS_HELPERS_KEYS: Record<AccountType, Record<number, CardHelperKeys>> = {
+  [AccountType.FORUM]: {
+    1: {
+      start: 'modal.identity_setup.forum.card_helper.step_1_start',
+      active: 'modal.identity_setup.forum.card_helper.step_1_active',
+      success: 'modal.identity_setup.forum.card_helper.step_1_success',
+      error: 'modal.identity_setup.forum.card_helper.step_1_error',
+    },
+    2: {
+      start: 'modal.identity_setup.forum.card_helper.step_2_start',
+      active: 'modal.identity_setup.forum.card_helper.step_2_active',
+      success: 'modal.identity_setup.forum.card_helper.step_2_success',
+    },
+    3: {
+      start: THREAD_URL,
+      active: THREAD_URL,
+      success: THREAD_URL,
+    },
+  },
+  [AccountType.DISCORD]: {},
+  [AccountType.TWITTER]: {},
+}
+
+const FORUM_CONNECT_STEPS_AMOUNT = 3
+
+const INITIAL_STATE: ModalState = {
+  currentType: ModalType.CHOOSE_ACCOUNT,
+  currentStep: 1,
+  isTimerActive: false,
+  isValidating: false,
+}
+
+function getAccountActionSteps(
+  account: AccountType,
+  stepsAmount: number,
+  icons: React.ReactNode[],
+  actions: (() => void)[],
+  currentStep: number,
+  t: ReturnType<typeof useFormatMessage>,
+  stepHelpers?: Record<number, string>
+): ActionCardProps[] {
+  if (stepsAmount === 0 || (icons.length !== stepsAmount && actions.length !== stepsAmount)) {
+    return []
+  }
+  return Array.from<unknown, ActionCardProps>({ length: stepsAmount }, (_, index) => {
+    const stepIdx = index + 1
+    const isDisabled = stepIdx > currentStep
+    const isCompleted = stepIdx < currentStep && stepIdx <= 1
+
+    let action = (
+      <Button basic onClick={actions[index]}>
+        {t(`modal.identity_setup.${account}.action_step_${stepIdx}`)}
+      </Button>
+    )
+
+    if (isCompleted) {
+      action = <CheckCircle size="24" outline />
+    }
+
+    if (isDisabled) {
+      action = <Lock />
+    }
+
+    return {
+      title: t(`modal.identity_setup.${account}.title_step_${stepIdx}`),
+      description: t(`modal.identity_setup.${account}.description_step_${stepIdx}`),
+      icon: icons[index],
+      action,
+      isDisabled,
+      helper: t(stepHelpers?.[stepIdx]),
+    }
+  })
+}
+
+function getHelperTexts(account: AccountType, stepsAmount: number): string[] {
+  if (stepsAmount === 0) {
+    return []
+  }
+  return Array.from({ length: stepsAmount }, (_, index) => `modal.identity_setup.${account}.helper_step_${index + 1}`)
+}
+
+function getTimeFormatted(totalSeconds: number) {
+  const minutes = Math.floor(totalSeconds / 60)
+  const seconds = totalSeconds % 60
+  return `${minutes}:${seconds < 10 ? `0${seconds}` : seconds}`
+}
+
+function AccountsConnectModal({ open, onClose }: ModalProps & { onClose: () => void }) {
```

Comment:
Isn't `onClose` part of `ModalProps`?

Reply:
Yes, but it must receive an event and data, which does not allow me to call the function simply as "onClose()", that's why I redefined it

![image](https://github.com/decentraland/governance/assets/45410089/ea9a6703-3423-4743-ae64-4807590bba74)